pkg/template/resource.go

```diff
@@ -110,8 +111,19 @@ func ApplyParamsToResourceTemplate(params []triggersv1.Param, rt json.RawMessage
 // applyParamToResourceTemplate returns the TriggerResourceTemplate with the
 // param value substituted for all matching param variables in the template
 func applyParamToResourceTemplate(param triggersv1.Param, rt json.RawMessage) json.RawMessage {
+	// The changes are for backward compatibility with both $(params) and $(tt.params)
+	// TODO(#606)
+	var (
+		paramsRegexp  = regexp.MustCompile(`\$\(params.(?P<var>[_a-zA-Z][_a-zA-Z0-9.-]*)\)`)
+		paramVariable string
+	)
+	templateParams := paramsRegexp.FindAllSubmatch(rt, -1)
 	// Assume the param is valid
-	paramVariable := fmt.Sprintf("$(params.%s)", param.Name)
+	if len(templateParams) == 0 {
+		paramVariable = fmt.Sprintf("$(tt.params.%s)", param.Name)
```

Comment:
Is there a reason we cannot attempt to replace with both `$(params)` as well as `$(tt.params)`?

Reply:
The reason is we will not be getting the information on user provided ResourceTemplate params
whether its `$(params)` or `$(tt.params)` because based on that we need to replace.
So currently with the help of regular expression able to distinguish whether its $(params) or $(tt.params).

Reply:
I don't think this works if a user mixes both params and tt.params in the same file. It should be safe to do a double pass - e.g.

```go
for _, tag := range []string{ "params", "tt.params"} {
  paramVariable := fmt.Sprintf("$(%s.%s)", tag, param.Name)
  rt = bytes.Replace(rt, []byte(paramVariable), []byte(paramValue), -1)
}
```